File: leetcode-go/Q051-Q100/Q080_Remove_Duplicates_from_Sorted_Array_II/Q080_Remove_Duplicates_from_Sorted_Array_II.go
package Q080_Remove_Duplicates_from_Sorted_Array_II

func removeDuplicates(nums []int) int {
	if len(nums) < 3 {
		return len(nums)
	}
	idx := 2
	for i := 2; i < len(nums); i++ {
		if !(nums[i] == nums[idx-1] && nums[i] == nums[idx-2]) {
			nums[idx] = nums[i]
			idx++
		}
	}
	return idx
}

File: src/Y2018/facebook/Q080_Remove_Duplicates_from_Sorted_Array_II.java
package Y2018.facebook;

public class Q080_Remove_Duplicates_from_Sorted_Array_II {

    public int removeDuplicates(int[] nums) {
        int cnt = 0, l = 0;
        for (int i = 0; i < nums.length; i++) {
            if (i == 0 || nums[i] == nums[i - 1]) {
                cnt++;
            } else {
                cnt = 1;
            }
            if (cnt == 1 || cnt == 2) {
                nums[l++] = nums[i];
            }
        }
        return l;
    }

}
